diff -urN tpm2-tools-3.0.1/tools/tpm2_nvread.c tpm2-tools-3.0.1-new/tools/tpm2_nvread.c
--- tpm2-tools-3.0.1/tools/tpm2_nvread.c	2017-12-11 08:34:01.000000000 -0700
+++ tpm2-tools-3.0.1-new/tools/tpm2_nvread.c	2017-12-15 18:08:18.361935382 -0700
@@ -120,6 +120,10 @@
         return false;
     }
 
+    if (max_data_size > MAX_NV_BUFFER_SIZE) {
+        max_data_size = MAX_NV_BUFFER_SIZE;
+    }
+
     UINT8 *data_buffer = malloc(data_size);
     if (!data_buffer) {
         LOG_ERR("oom");
diff -urN tpm2-tools-3.0.1/tools/tpm2_nvwrite.c tpm2-tools-3.0.1-new/tools/tpm2_nvwrite.c
--- tpm2-tools-3.0.1/tools/tpm2_nvwrite.c	2017-12-11 08:34:01.000000000 -0700
+++ tpm2-tools-3.0.1-new/tools/tpm2_nvwrite.c	2017-12-15 18:08:09.055066735 -0700
@@ -116,6 +116,10 @@
         return false;
     }
 
+    if (max_data_size > MAX_NV_BUFFER_SIZE) {
+        max_data_size = MAX_NV_BUFFER_SIZE;
+    }
+
     UINT16 data_offset = 0;
     UINT16 bytes_left = ctx.nv_buffer.t.size;
     while (bytes_left > 0) {
