diff -ur tpm2.0-tools-1.1.0/src/tpm2_rc_decode.c tpm2.0-tools-1.1.0-new/src/tpm2_rc_decode.c
--- tpm2.0-tools-1.1.0/src/tpm2_rc_decode.c	2016-11-04 07:13:32.000000000 -0700
+++ tpm2.0-tools-1.1.0-new/src/tpm2_rc_decode.c	2017-04-05 14:23:36.654804125 -0700
@@ -246,6 +246,7 @@
         fprintf (stderr, "Unknown TPM_RC format\n");
         return -1;
     }
+    return 0;
 }
 /* Top level function to dump human readable data about TPM_RCs.
  */
